src/components/HeartFavorite/HeartFavorite.container.js

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import propTypes from 'prop-types';
+
+import { createFavorite } from 'store/actions/favoritesActions';
+import { isSongInFavorites } from 'utils/favorites';
+
+import HeartFavoriteComponent from './HeartFavorite.component';
+
+const HeartFavoriteContainer = ({ fileName, path }) => {
+  const [isFavorite, setIsFavorite] = useState(false);
+
+  const favorites = useSelector((state) => state.favorites.get('songs'));
+  const dispatch = useDispatch();
+
+  const handleFavorite = () => {
+    dispatch(createFavorite(fileName, path));
+  };
+
+  useEffect(() => {
+    setIsFavorite(isSongInFavorites(favorites, path));
```

Comment:
I don't think you need a state for this. Also, if the song is not in favorites you are forcing a useless state update.